plugins/modules/packaging/os/rpm_ostree_pkg.py

```diff
@@ -0,0 +1,176 @@
+    description: State changes.
+    returned: always
+    type: bool
+    sample: True
+action:
+    description: Action performed.
+    returned: always
+    type: str
+    sample: 'install'
+packages:
+    description: A list of packages specified.
+    returned: always
+    type: list
+    sample: ['nfs-utils']
+stdout:
+    description: Stdout of rpm-ostree command.
+    returned: always
+    type: str
+    sample: 'Staging deployment...done\n...'
+stderr:
+    description: Stderr of rpm-ostree command.
+    returned: always
+    type: str
+    sample: ''
+cmd:
+    description: Full command used for performed action.
+    returned: always
+    type: str
+    sample: 'rpm-ostree uninstall --allow-inactive --idempotent --unchanged-exit-77 nfs-utils'
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+class RpmOstreePkg:
+    def __init__(self, module):
+        self.module = module
+        self.params = module.params
+        self.state = module.params['state']
+
+    def ensure(self):
+        results = dict(
+            rc=0,
+            changed=False,
+            action='',
+            packages=[],
+            stdout='',
+            stderr='',
+            cmd='',
+        )
+
+        # Ensure rpm-ostree command exists
+        cmd = [self.module.get_bin_path('rpm-ostree', required=True)]
+
+        # Decide action to perform
+        if self.state in ('present'):
+            results['action'] = 'install'
+            cmd.append('install')
+        elif self.state in ('absent'):
+            results['action'] = 'uninstall'
+            cmd.append('uninstall')
+
+        # Additional parameters
+        cmd.extend(['--allow-inactive', '--idempotent', '--unchanged-exit-77'])
+        for pkg in self.params['name']:
+            cmd.append(pkg)
+            results['packages'].append(pkg)
+
+        rc, out, err = (0, '', '')
+        if not self.module.check_mode:
+            rc, out, err = self.module.run_command(cmd)
+
+        results.update(dict(
+            rc=rc,
+            cmd=' '.join(cmd),
+            stdout=out,
+            stderr=err,
+        ))
+
+        # A few possible options:
+        #     - rc=0  - succeeded in making a change
+        #     - rc=77 - no change was needed
+        #     - rc=?  - error
+        if rc == 0:
+            results['changed'] = True
+        elif rc == 77:
+            results['changed'] = False
+            results['rc'] = 0
+        else:
+            self.module.fail_json(msg='non-zero return code', **results)
+
+        self.module.exit_json(**results)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(
+                default="present",
+                choices=['absent', 'present']
+            ),
+            name=dict(
+                aliases=["pkg"],
+                required=True,
+                type='list',
+                elements='str',
+            ),
+        ),
+        supports_check_mode=True
```

Comment:
```suggestion
        supports_check_mode=False
```
from the discussion above